apps/server/src/modules/user-login-migration/uc/close-user-login-migration.uc.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
+import { EntityId, Permission, User, UserLoginMigrationDO } from '@shared/domain';
+import { Action, AuthorizationService } from '@src/modules/authorization';
+import {
+	UserLoginMigrationGracePeriodExpiredLoggableException,
+	UserLoginMigrationNotFoundLoggableException,
+} from '../error';
+import { SchoolMigrationService, UserLoginMigrationRevertService, UserLoginMigrationService } from '../service';
+
+@Injectable()
+export class CloseUserLoginMigrationUc {
+	constructor(
+		private readonly userLoginMigrationService: UserLoginMigrationService,
+		private readonly schoolMigrationService: SchoolMigrationService,
+		private readonly userLoginMigrationRevertService: UserLoginMigrationRevertService,
+		private readonly authorizationService: AuthorizationService
+	) {}
+
+	async closeMigration(userId: EntityId, schoolId: EntityId): Promise<UserLoginMigrationDO> {
+		const userLoginMigration: UserLoginMigrationDO | null = await this.userLoginMigrationService.findMigrationBySchool(
+			schoolId
+		);
+
+		if (!userLoginMigration) {
+			throw new UserLoginMigrationNotFoundLoggableException(schoolId);
+		}
+
+		const user: User = await this.authorizationService.getUserWithPermissions(userId);
+		this.authorizationService.checkPermission(user, userLoginMigration, {
+			requiredPermissions: [Permission.USER_LOGIN_MIGRATION_ADMIN],
+			action: Action.write,
+		});
+
+		if (userLoginMigration.finishedAt && Date.now() >= userLoginMigration.finishedAt.getTime()) {
```

Comment:
You could seperate these things in private functions for better readability..
e.g. if (isMigrationFinishedBeforeNow) { 
...
else if (isMigrationAlreadyClosed())
... else {
   closeMigration()...
}